gson/src/main/java/com/google/gson/ToNumberPolicy.java

```diff
@@ -68,22 +68,30 @@ public Number readNumber(JsonReader in) throws IOException {
     @Override
     public Number readNumber(JsonReader in) throws IOException, JsonParseException {
       String value = in.nextString();
-      try {
-        return Long.parseLong(value);
-      } catch (NumberFormatException longE) {
+      if (value.contains(".")) {
```

Comment:
Could also use `indexOf('.') >= 0` here, which might be more efficient due to using a `char`. But I am not sure if that is really more efficient and if that is worth it.

Reply:
@Marcono1234 the difference between `contains(".")` and `indexOf('.') >= 0` is very small, and nearly negligible. But you're right, `indexOf` is slightly faster. I just pushed the tweak.